packages/server/modules/workspaces/tests/integration/projects.graph.spec.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  BasicTestWorkspace,
+  createTestWorkspace
+} from '@/modules/workspaces/tests/helpers/creation'
+import {
+  BasicTestUser,
+  createAuthTokenForUser,
+  createTestUser
+} from '@/test/authHelper'
+import {
+  CreateWorkspaceProjectDocument,
+  GetWorkspaceProjectsDocument
+} from '@/test/graphql/generated/graphql'
+import {
+  createTestContext,
+  testApolloServer,
+  TestApolloServer
+} from '@/test/graphqlHelper'
+import { beforeEachContext } from '@/test/hooks'
+import { Roles } from '@speckle/shared'
+import { expect } from 'chai'
+import cryptoRandomString from 'crypto-random-string'
+
+describe('Workspace project GQL CRUD', () => {
+  let apollo: TestApolloServer
+
+  const testUser: BasicTestUser = {
+    id: '',
+    name: 'John Speckle',
+    email: 'john-speckle-workspace-project-admin@example.org',
+    role: Roles.Server.Admin
+  }
+
+  const testNonWorkspaceMemberUser: BasicTestUser = {
+    id: '',
+    name: 'John Nobody',
+    email: 'john-nobody@example.org',
+    role: Roles.Server.User
+  }
+
+  before(async () => {
+    await beforeEachContext()
+    await Promise.all(
+      [testUser, testNonWorkspaceMemberUser].map((user) => createTestUser(user))
+    )
+    const token = await createAuthTokenForUser(testUser.id, AllScopes)
+    apollo = await testApolloServer({
+      context: createTestContext({
+        auth: true,
+        userId: testUser.id,
+        token,
+        role: testUser.role,
+        scopes: AllScopes
+      })
+    })
+  })
+
+  describe('when specifying a workspace id during project creation', () => {
+    let workspace: BasicTestWorkspace
+
+    before(async () => {
+      const testWorkspace: BasicTestWorkspace = {
+        id: '',
+        ownerId: testUser.id,
+        name: 'My Test Workspace'
+      }
+
+      await createTestWorkspace(testWorkspace, testUser)
+
+      workspace = testWorkspace
+    })
+
+    it('should create the project in that workspace', async () => {
+      const projectName = cryptoRandomString({ length: 6 })
+
+      const createRes = await apollo.execute(CreateWorkspaceProjectDocument, {
+        input: {
+          name: projectName,
+          workspaceId: workspace.id
+        }
+      })
+
+      const getRes = await apollo.execute(GetWorkspaceProjectsDocument, {
+        id: workspace.id
+      })
+
+      const workspaceProject = getRes.data?.workspace.projects.items.find(
+        (project) => project.name === projectName
+      )
+
+      expect(createRes).to.not.haveGraphQLErrors()
+      expect(getRes).to.not.haveGraphQLErrors()
+      expect(workspaceProject).to.exist
+    })
+  })
+
+  describe('when querying workspace projects', () => {
+    let workspace: BasicTestWorkspace
+
+    before(async () => {
+      const testWorkspace: BasicTestWorkspace = {
+        id: '',
+        ownerId: testUser.id,
+        name: 'My Test Workspace'
+      }
+
+      await createTestWorkspace(testWorkspace, testUser)
+
+      workspace = testWorkspace
```

Comment:
Here like in the other PR, there's no need to have 2 variables for the same workspace - testWorkspace and workspace. Just move `testWorkspace` outside of `before()` and use that.

Reply:
much better, no sure why I did it this way